core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -747,9 +747,7 @@ void Cuda::impl_initialize(const Cuda::SelectDevice config,
                            size_t /*num_instances*/) {
   Impl::CudaInternal::singleton().initialize(config.cuda_device_id, 0);
 
-#if defined(KOKKOS_ENABLE_PROFILING)
   Kokkos::Profiling::initialize();
-#endif
```

Comment:
Why were we initializing here in the first place?

Reply:
That was added by @crtrott here: https://github.com/kokkos/kokkos/commit/fae0d00e74a355615fa47a9a323b056dfae76fc2 , apparently it addressed this issue https://github.com/kokkos/kokkos/issues/192 , where Trilinos used to initialize execution spaces individually rather than initializing Kokkos. If we're going to change that, I'd prefer it be in another PR